### PR TITLE
Issues/7675 refactor loginfields

### PR DESCRIPTION
**Pull request by aerych**
Refs #7675
This does a bit of refactor to the LoginFields class to clarify its usage in certain scenarios and better organize some of its fields.  Doing this in advance of adding some new functionality to support social login.

To test:
Test the following auth flows:
- wpcom + magic link
- wpcom + password
- self-hosted
- saved safari credentials (iOS 10 only)
- 1password
- Sign up

Confirm that all still succeed. Confirm the epilogue still shows shows site info when its available.
Run the test suite. Confirm that NUX related tests pass.  In particular the ones for SigninHelpers and the LoginFacade.

Needs review: @nheagy

### CODE REVIEW

File: WordPress/Classes/ViewRelated/NUX/LoginFields.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    /// Stores the user's account identifier (either email address or username) that is
+    /// entered in the login flow. By convention, even if the user is logging in
+    /// via an email address this field should store that value.
+    var username = ""
+
+    /// The user's password.
+    var password = ""
+
+    /// The site address if logging in via the self-hosted flow.
+    var siteAddress = ""
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/LoginFields.swift]: I was debating if `siteAddress` should be optional (and maybe `multifactorCode` and `emailAddress`), but I think this is something to leave for now. Just something to keep in mind for the future.

File: WordPress/Classes/ViewRelated/NUX/LoginFields.swift [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    /// The two factor code entered by a user.
+    var multifactorCode = "" // 2fa code
+
+    /// Used by the SignupViewController. Signup currently asks for both a
+    /// username and an email address.  This can be factored away when we revamp
+    /// the signup flow.
+    var emailAddress = ""
+
+    var meta = LoginFieldsMeta()
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/LoginFields.swift]: I had to think about `meta` as a property name. But, hmm, I think I like it 😉

File: WordPress/Classes/ViewRelated/NUX/LoginFields.swift (hunk 1 above)
Comment (aerych) [reply WordPress/Classes/ViewRelated/NUX/LoginFields.swift]: I actually started to go with it as an optional, but we're already checking for `empty` in several places.  This seemed like a lighter touch.  Def something we can revisit.

File: WordPress/Classes/ViewRelated/NUX/LoginFields.swift (hunk 2 above)
Comment (aerych) [reply WordPress/Classes/ViewRelated/NUX/LoginFields.swift]: I originally had something like `metaFields` or `metaData` or something like that.  Seemed a bit silly when I stared at it for a bit so... yeah... meta. 🤷‍♂️

**Comment by aerych** [comment]
Thanks @nheagy!!  I pushed a small code style tweak to silence hound.  I'll merge after build checks complete.

**Pull request merged**